client/src/components/SplitList.jsx

```diff
@@ -1,100 +1,133 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import IndividualOwes from './IndividualOwes.jsx';
+import axios from 'axios';
+// import Dotenv from 'dotenv-webpack';
 
 const SplitList = (props) => {
+// change this to hold its own state for tip?
+
   const numberOfGuests = props.guests.length;
   const billTotalWithoutTipOrTax = props.totalCost;
   const tipPercentage = props.tipPercentage;
   const tipAmount = billTotalWithoutTipOrTax * tipPercentage / 100;
   const billWithTip = billTotalWithoutTipOrTax + tipAmount;
   // API CALL FOR TIP PERCENTAGE
-  const taxPercentage = 7;
-
-  const taxAmount = billTotalWithoutTipOrTax * taxPercentage / 100;
-  const billWithTipAndTax = billWithTip + taxAmount;
-
-  let paymentsOwed;
+  let taxPercentage;
 
-  const getGuestTotals = (guests) => {
-    const totals = {};
-    const guestArray = guests;
-    const guestIDs = [];
-    for (let i = 0; i < guestArray.length; i++) {
-      const currentGuestID = guestArray[i].guestName;
-      // const currentGuestID = guestArray[i].guest_id;
-      guestIDs.push(currentGuestID);
-      totals[currentGuestID] = 0;
-      const currentGuestOrders = guestArray[i].order;
-      for (let j = 0; j < currentGuestOrders.length; j++) {
-        const orderItemCost = currentGuestOrders[j].price;
-        const howManyOrdered = 1;
-        // const howManyOrdered = currentGuestOrders[j].qty;
-        const itemTotal = orderItemCost * howManyOrdered;
-        totals[currentGuestID] += itemTotal;
+  useEffect(() => {
+    console.log('💸 💸 💸 💸 💸 SPENT A PENNY 💸 💸 💸 💸 💸 💸 💸 ');
+    const options = {
+      method: 'GET',
+      url: `https://u-s-a-sales-taxes-per-zip-code.p.rapidapi.com/${props.zipCode}`,
+      headers: {
+        'x-rapidapi-host': process.env.TAXAPIHOST,
+        'x-rapidapi-key': process.env.TAXAPIKEY
       }
     };
 
-    for (const guestID in totals) {
-      const tip = totals[guestID] * (tipPercentage / 100);
-      const tax = totals[guestID] * (taxPercentage / 100);
-      totals[guestID] = (totals[guestID] + tip + tax).toFixed(2);
-    }
+    axios.request(options).then(function (response) {
```

Comment:
You don't need to make this call more than once on load right? I would put this part in its own useEffect with only the `props.zipCode` as a dependency, so it will only run when the zipCode changes. RIght now, it will run if you switch the split type when it will always return the same value.

```
useEffect(() => {
  // do all that stuff 
}, [props.zipCode]);
```